src/models/ScoreLog.php

```diff
@@ -69,28 +75,49 @@ private static function scorelogFromRow(Map<string, string> $row): ScoreLog {
     int $level_id,
     int $team_id,
     bool $any_team,
+    bool $refresh = false,
   ): Awaitable<bool> {
     $db = await self::genDb();
-
-    if ($any_team) {
-      $result =
-        await $db->queryf(
-          'SELECT COUNT(*) FROM scores_log WHERE level_id = %d AND team_id IN (SELECT id FROM teams WHERE id != %d AND visible = 1)',
-          $level_id,
-          $team_id,
-        );
-    } else {
-      $result =
-        await $db->queryf(
-          'SELECT COUNT(*) FROM scores_log WHERE level_id = %d AND team_id = %d',
-          $level_id,
-          $team_id,
-        );
+    $mc_result = self::getMCRecords('LEVEL_CAPTURES');
+    if ((!$mc_result) || (count($mc_result) === 0) || ($refresh)) {
+      $level_captures = Map {};
+      $result = await $db->queryf('SELECT level_id, team_id FROM scores_log');
+      foreach ($result->mapRows() as $row) {
+        if ($level_captures->contains(intval($row->get("level_id")))) {
+          $level_capture_teams =
+            $level_captures->get(intval($row->get("level_id")));
+          /* HH_IGNORE_ERROR[4064] */
+          $level_capture_teams->add(intval($row->get("team_id")));
+          $level_captures->set(
+            intval($row->get("level_id")),
+            $level_capture_teams,
+          );
+        } else {
+          $level_capture_teams = Vector {};
+          $level_capture_teams->add(intval($row->get("team_id")));
+          $level_captures->add(
+            Pair {intval($row->get("level_id")), $level_capture_teams},
+          );
+        }
+      }
+      self::setMCRecords('LEVEL_CAPTURES', new Map($level_captures));
     }
-
-    if ($result->numRows() > 0) {
-      invariant($result->numRows() === 1, 'Expected exactly one result');
-      return intval($result->mapRows()[0]['COUNT(*)']) > 0;
+    $level_captures = self::getMCRecords('LEVEL_CAPTURES');
+    /* HH_IGNORE_ERROR[4062] */
```

Comment:
You don't have to specify a comment for each of these `HH_IGNORE_ERROR`s if they're all the same. Just do one. Hack has an `/* UNSAFE_BLOCK */`, but that's caused me more problems than it's solved.